server/channels/jobs/jobs.go

```diff
@@ -95,17 +95,17 @@ func (srv *JobServer) GetJob(c request.CTX, id string) (*model.Job, *model.AppEr
 	return job, nil
 }
 
-func (srv *JobServer) ClaimJob(job *model.Job) (bool, *model.AppError) {
-	updated, err := srv.Store.Job().UpdateStatusOptimistically(job.Id, model.JobStatusPending, model.JobStatusInProgress)
+func (srv *JobServer) ClaimJob(job *model.Job) (*model.Job, *model.AppError) {
+	newJob, err := srv.Store.Job().UpdateStatusOptimistically(job.Id, model.JobStatusPending, model.JobStatusInProgress)
 	if err != nil {
-		return false, model.NewAppError("ClaimJob", "app.job.update.app_error", nil, "", http.StatusInternalServerError).Wrap(err)
+		return nil, model.NewAppError("ClaimJob", "app.job.update.app_error", nil, "", http.StatusInternalServerError).Wrap(err)
 	}
 
-	if updated && srv.metrics != nil {
-		srv.metrics.IncrementJobActive(job.Type)
+	if newJob != nil && srv.metrics != nil {
```

Comment:
Can there be a case where err == nil and newJob is also nil? I see a lot of cases where `if !claimed` is being replaced with `if job == nil`. Do we still need that?

Reply:
@agnivade if `RowsAffected()` returns != 1, we return `nil, nil`, that's the case.  Just double-checked: all calls to `UpdateStatusOptimistically` check for `job == nil` (as they should, unless we make a change to the RowsAffected logic).

Reply:
We can change this if we want, it just looks like how we've always done it.

Reply:
>  (as they should, unless we make a change to the RowsAffected logic).

I think we can return a dedicated error if RowsAffected() is != 1. Something like errors.New("more than 1 row affected"). Then the error handling at the caller side becomes simple. Will leave it to you if you want to do it in this PR or separately.